v2/internal/testcommon/samples_tester.go

```diff
@@ -211,12 +211,13 @@ func (t *SamplesTester) setOwnershipAndReferences(samples map[string]genruntime.
 		}
 
 		if ownersName != "" {
-			var err error
 			sample = setOwnersName(sample, ownersName)
-			sample, err = t.updateARMReferencesForTest(sample)
-			if err != nil {
-				return err
-			}
+		}
+
+		var err error
+		sample, err = t.updateARMReferencesForTest(sample)
```

Comment:
Putting it outside the check. Since the method was not called when the owner is not RG and neither we're using random names.

Reply:
We were only updating references for resources owned by RG _and_ were using random names.